### ISSUE TITLE
Doesn't work with 3rd level domains

### ISSUE THREAD

**Issue by Milan-Benes** [silkeh/pdns_api.sh#1]
pdns_api.sh assumes only 2nd level domains. This is my attempt at generalization:
```
--- pdns_api.sh.orig    2016-07-18 20:47:54.563236132 +0200
+++ pdns_api.sh.new     2016-07-18 21:00:03.432873377 +0200
@@ -60,11 +60,15 @@ setup() {
   # Domain and token from arguments
   domain="${1}"
   token="${2}"
+  zone=""
 
   # Create the zone name from the arguments
   IFS='.' read -a domain_array <<< "$domain"
-  zone="${domain_array[*]: -2:1}.${domain_array[*]: -1:1}"
-  echo $zone
+  domlen=${#domain_array[@]}
+  for (( i=1; i<${domlen}; i++ ));
+  do
+    zone="${zone}${domain_array[$i]}."
+  done
 
   # Record name
   name="_acme-challenge.$domain"
```

**Comment by silkeh** [comment]
Thanks for the suggestion.
The problem with this is that the zone isn't necessarily one level down from the requested domain.
Take for example (`simple.test.example.com`), this host could be in `.test.example.com` (your code works, mine doesn't) or in `.example.com` (current code works, your code doesn't) or even in `.com` (neither work).

I will take a look at requesting the zones from the API later this week.

**Comment by Milan-Benes** [comment]
Hello,
I see your point. My solution is short and nice, but wrong/incomplete :). Maybe the correct approach is to query the nameserver for all possible zones and then query those zones for the complementary records (A and CNAME) and inserting the challenge where appropriate? I mean querying for zones test.example.com., example.com. and com. and then searching for simple, simple.test and simple.test.example respectively. It is complicated but hopefully it could cover the issue more completely...

**Comment by okket** [comment]
My take on this problem, see #3. It's for the case that subdomains are separate zones in PowerDNS.

**Comment by silkeh** [comment]
Closed with #3.
